CI/check_boost_test_macro.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+test_string="BOOST_TEST"
+grep $test_string -n -r Tests --include "*.cpp" --include "*.hpp"
```

Comment:
I'm not sure if we actually need to search `*.hpp` here, but if we do, we probably want to search `*.ipp` too.

Reply:
In some cases I believe we have a few hpp files that included as helpers in multiple tests. That mostly got moved to separate compilation units / cpp files. 

I'll add ipp to be safe.